src/renderer/components/recording-page/recording-page.test.tsx

```diff
@@ -0,0 +1,134 @@
+
+import 'web-audio-test-api';
+import { noOp } from '../../../common/env-and-consts';
+import mediaService from '../../services/media';
+import { readWavAsBlob } from '../../utils/media-utils';
+
+import RecordingPage from './index';
+
+// Mock the remote module to prevent import failure.
+jest.mock('@electron/remote', () => ({ dialog: jest.fn() }));
+
+jest.mock('../../utils/media-utils');
+
+// Automatic mock doesn't work because the import in the module causes error when in a testing environment
+jest.mock('../../services/media', () => ({
+  switchOnAudioInput: jest.fn(noOp),
+  stopRecording: jest.fn(noOp),
+  startRecording: jest.fn(noOp),
+  audioBlobToWavArrayBuffer: jest.fn(noOp),
+  createNewAudioGraph: jest.fn(noOp),
+  playBlob: jest.fn(noOp),
+  createOscillator: jest.fn(noOp),
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  playAudioNode: jest.fn((node: AudioScheduledSourceNode | AudioNode) => {
+    return new Promise(noOp);
+  }),
+  stopPlaying: jest.fn(noOp),
+}));
+
+jest.mock;
+
+describe('recordingPage', () => {
+  it('starts recording when r is pressed', async () => {
+    expect.assertions(4);
+
+    // Fix for 'Reference Error: __static is not defined
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (global as any).__static = '';
+
+    //Define mocks to confirm that calls have been made
+    const alertStartRecording = jest.spyOn(mediaService, 'startRecording');
+    const alertAudioBlobToWavArrayBuffer = jest.spyOn(mediaService, 'audioBlobToWavArrayBuffer');
+    const stopRecording = jest.spyOn(mediaService, 'stopRecording');
+    const switchOnAudio = jest.spyOn(mediaService, 'switchOnAudioInput');
+
+    render(<RecordingPage onBack={noOp} recordingItems={[]} basePath={''} scaleKey={'C'} octave={2} />);
+    //R is 82
+    fireEvent.keyDown(document, { key: 'R', code: '82' });
+    await new Promise((r) => setTimeout(r, 3000));
+    fireEvent.keyUp(document, { key: 'R', code: '82' });
+
+    expect(alertStartRecording).toHaveBeenCalledTimes(1);
+    expect(alertAudioBlobToWavArrayBuffer).toHaveBeenCalledTimes(1);
+    expect(stopRecording).toHaveBeenCalledTimes(1);
+    expect(switchOnAudio).toHaveBeenCalledTimes(1);
+  });
+
+  it('plays a tone when s is pressed', async () => {
+    expect.assertions(3);
+
+    // Fix for 'Reference Error: __static is not defined
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (global as any).__static = '';
+    const audioContext = new AudioContext();
+    const osc = audioContext.createOscillator();
+
+    //Define mocks to confirm that calls have been made
+    const createNewAudioGraph = jest.spyOn(mediaService, 'createNewAudioGraph');
+    const createOscillator = mocked(mediaService.createOscillator);
+    osc.start();
+    createOscillator.mockReturnValue(osc);
+
+    const playAudioNode = mocked(mediaService.playAudioNode);
+    playAudioNode.mockResolvedValue(new Promise(noOp));
```

Comment:
What is this trying to accomplish? A `Promise` that doesn't call `resolve` is a broken Promise that never resolves.

Reply:
I was trying to mock the method to return a Promise where resolve didn't do anything - is that not the way to go about this?

Reply:
`playAudioNode()` returns `Promise<void>` which means its resolved value is a `void` - here you are saying the resolved value is a `Promise`. That's disconnect No. 1.

`Promise<void>` should be `new Promise(resolve => resolve())`. Passing `noOp` means `resolve` will never be called, so the Promise will never resolve. That's disconnect No. 2.